### ISSUE TITLE
Exception thrown for BeginTransaction together WithOptions

### ISSUE THREAD

**Issue by madness2222** [ThatRendle/Simple.Data#223]
Hi,

Trying v0.18 WithOptions at the end of my OpenNamedConnection. When I tried calling BeginTransaction I get InvalidOperationException error with the message "Operation is not valid due to the current state of the object.".

If I take out WithOptions at the end of my OpenNamedConnection, BeginTransaction is successful created and able to carry on doing what I need to do inside the transaction.

Many Thanks.

**Comment by ThatRendle** [comment]
I will fix this, but ideally, you want to begin the transaction, then call WithOptions, something like:

``` csharp
var db = Database.Open();
var tx = db.BeginTransaction().WithOptions(new AdoOptions(identityInsert: true));
tx.People.Insert(Name: "Frak");
tx.Commit();
```

**Comment by madness2222** [comment]
Ah, Ok. I'll give that a go as I was following the example in the test project to see how it is done.

**Comment by ThatRendle** [comment]
Fixed for 0.18.1 (1.0 RC2).
